### PR TITLE
Add cosign signing and verification for modules

### LINKED ISSUE #25: Integrate with Cosign for signing and verifying modules
Given that Cosign complains when using image tags, we could shell-out to the cosign CLI from `timoni mod push` and `timoni mod pull` and pass the artifact digest URL. The push and pull commands could have cosign flags similar to [nerdctl](https://github.com/containerd/nerdctl/blob/main/docs/cosign.md).

### PR DESCRIPTION
**Pull request by loktev-d**
Add Cosgin integration for image signing to close #25

Flags for `mod push`
```
--sign=cosign Signs the module with Cosign
--cosign-key  The Cosign private key for signing the module
```

Flags for `mod pull`
```
--certificate-identity string             The identity expected in a valid Fulcio certificate for verifying the Cosign signature.
                                          Valid values include email address, DNS names, IP addresses, and URIs.
                                          Either --certificate-identity or --certificate-identity-regexp must be set for keyless flows.
--certificate-identity-regexp string      A regular expression alternative to --certificate-identity for verifying the Cosign signature.
                                          Accepts the Go regular expression syntax described at https://golang.org/s/re2syntax.
                                          Either --certificate-identity or --certificate-identity-regexp must be set for keyless flows.
--certificate-oidc-issuer string          The OIDC issuer expected in a valid Fulcio certificate for verifying the Cosign signature,
                                          e.g. https://token.actions.githubusercontent.com or https://oauth2.sigstore.dev/auth.
                                          Either --certificate-oidc-issuer or --certificate-oidc-issuer-regexp must be set for keyless flows.
--certificate-oidc-issuer-regexp string   A regular expression alternative to --certificate-oidc-issuer for verifying the Cosign signature.
                                          Accepts the Go regular expression syntax described at https://golang.org/s/re2syntax.
                                          Either --certificate-oidc-issuer or --certificate-oidc-issuer-regexp must be set for keyless flows.
--verify=cosign                           Verifies the signed module with Cosign.
--cosign-key string                       The Cosign public key for verifying the module.
```

### CODE REVIEW

File: internal/signutil/signutil.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+func Sign(log logr.Logger, imageRef string, keyRef string) error {
+	cosignExecutable, err := exec.LookPath("cosign")
+	if err != nil {
+		return fmt.Errorf("executing cosign failed: %w", err)
+	}
+
+	cosignCmd := exec.Command(cosignExecutable, []string{"sign"}...)
+	cosignCmd.Env = os.Environ()
+
+	// if key is empty, use keyless mode(experimental)
```
Comment (stefanprodan) [inline internal/signutil/signutil.go:40]: Keyless is no longer experimental since Cosign v2. Do we really need `COSIGN_EXPERIMENTAL=true`?

Comment (loktev-d) [reply internal/signutil/signutil.go:40]: I added it for older Cosign versions compatibility, but if it's not that important, I can remove it

Comment (stefanprodan) [reply internal/signutil/signutil.go:40]: The old one v1, didn’t had the `—yes` flag so it would fail anyway

**Review by stefanprodan — COMMENTED** [review]
@loktev-d can you please change the flags to `--sign=cosign` and `--verify=cosign`, to make room in the future for alternative signing tools.

File: internal/signutil/signutil.go (hunk 1 above)
Comment (loktev-d) [reply internal/signutil/signutil.go:40]: Yep, just checked it, I guess I just remove it then

**Comment by loktev-d** [comment]
@stefanprodan Done

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @loktev-d 🥇 

PS. I'll add docs and e2e tests in a followup PR.

**Comment by stefanprodan** [comment]
Going to open an issue for extending the verification to modules referred in bundles. It could be a dedicated command e.g.  `timoni bundle verify` but I'm undecided how to go about it, as each module can have it's own pub key or issuer.

**Pull request merged**